### PR TITLE
config: add function ensuring promise resolution

### LINKED ISSUE #16: An in-range update of mocha is breaking the build 🚨
## The devDependency [mocha](https://github.com/mochajs/mocha) was updated from `6.1.4` to `6.2.0`.

🚨 [View failing branch](https://github.com/CloudNativeJS/cloud-health/compare/master...CloudNativeJS:greenkeeper%2Fmocha-6.2.0).

This version is **covered** by your **current version range** and after updating it in your project **the build failed**.

mocha is a devDependency of this project. It **might not break your production code or affect downstream projects**, but probably breaks your build or test tools, which may **prevent deploying or publishing**.

<details>
<summary>Status Details</summary>

- ❌ **continuous-integration/travis-ci/push:** The Travis CI build failed ([Details](https://travis-ci.org/CloudNativeJS/cloud-health/builds/560729063?utm_source=github_status&utm_medium=notification)).
</details>

---

<details>
<summary>Release Notes for v6.2.0</summary>

<h1>6.2.0 / 2019-07-18</h1>
<h2><g-emoji class="g-emoji" alias="tada" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f389.png">🎉</g-emoji> Enhancements</h2>
<ul>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3827" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3827/hovercard">#3827</a>: Do not fork child-process if no Node flags are present (<a href="https://urls.greenkeeper.io/boneskull"><strong>@boneskull</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3725" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3725/hovercard">#3725</a>: Base reporter store ref to console.log, see <a href="https://urls.greenkeeper.io/mochajs/mocha/wiki/HOW-TO:-Correctly-stub-stdout">mocha/wiki</a> (<a href="https://urls.greenkeeper.io/craigtaub"><strong>@craigtaub</strong></a>)</li>
</ul>
<h2><g-emoji class="g-emoji" alias="bug" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f41b.png">🐛</g-emoji> Fixes</h2>
<ul>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3942" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3942/hovercard">#3942</a>: Fix "No test files found" Error when file is passed via <code>--file</code> (<a href="https://urls.greenkeeper.io/gabegorelick"><strong>@gabegorelick</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3914" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3914/hovercard">#3914</a>: Modify Mocha constructor to accept options <code>global</code> or <code>globals</code> (<a href="https://urls.greenkeeper.io/pascalpp"><strong>@pascalpp</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3894" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3894/hovercard">#3894</a>: Fix parsing of config files with <code>_mocha</code> binary (<a href="https://urls.greenkeeper.io/juergba"><strong>@juergba</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3834" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3834/hovercard">#3834</a>: Fix CLI parsing with default values (<a href="https://urls.greenkeeper.io/boneskull"><strong>@boneskull</strong></a>, <a href="https://urls.greenkeeper.io/juergba"><strong>@juergba</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3831" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3831/hovercard">#3831</a>: Fix <code>--timeout</code>/<code>--slow</code> string values and duplicate arguments (<a href="https://urls.greenkeeper.io/boneskull"><strong>@boneskull</strong></a>, <a href="https://urls.greenkeeper.io/juergba"><strong>@juergba</strong></a>)</li>
</ul>
<h2><g-emoji class="g-emoji" alias="book" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4d6.png">📖</g-emoji> Documentation</h2>
<ul>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3906" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3906/hovercard">#3906</a>: Document option to define custom report name for XUnit reporter (<a href="https://urls.greenkeeper.io/pkuczynski"><strong>@pkuczynski</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3889" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3889/hovercard">#3889</a>: Adds doc links for mocha-examples (<a href="https://urls.greenkeeper.io/craigtaub"><strong>@craigtaub</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3887" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3887/hovercard">#3887</a>: Fix broken links (<a href="https://urls.greenkeeper.io/toyjhlee"><strong>@toyjhlee</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3841" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3841/hovercard">#3841</a>: Fix anchors to configuration section (<a href="https://urls.greenkeeper.io/trescube"><strong>@trescube</strong></a>)</li>
</ul>
<h2><g-emoji class="g-emoji" alias="mag" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f50d.png">🔍</g-emoji> Coverage</h2>
<ul>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3915" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3915/hovercard">#3915</a>, <a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3929" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3929/hovercard">#3929</a>: Increase tests coverage for <code>--watch</code> options (<a href="https://urls.greenkeeper.io/geigerzaehler"><strong>@geigerzaehler</strong></a>)</li>
</ul>
<h2><g-emoji class="g-emoji" alias="nut_and_bolt" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f529.png">🔩</g-emoji> Other</h2>
<ul>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3953" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3953/hovercard">#3953</a>: Collect test files later, prepares improvements to the <code>--watch</code> mode behavior (<a href="https://urls.greenkeeper.io/geigerzaehler"><strong>@geigerzaehler</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3939" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3939/hovercard">#3939</a>: Upgrade for npm audit (<a href="https://urls.greenkeeper.io/boneskull"><strong>@boneskull</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3930" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3930/hovercard">#3930</a>: Extract <code>runWatch</code> into separate module (<a href="https://urls.greenkeeper.io/geigerzaehler"><strong>@geigerzaehler</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3922" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3922/hovercard">#3922</a>: Add <code>mocha.min.js</code> file to stacktrace filter (<a href="https://urls.greenkeeper.io/brian-lagerman"><strong>@brian-lagerman</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3919" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3919/hovercard">#3919</a>: Update CI config files to use Node-12.x (<a href="https://urls.greenkeeper.io/plroebuck"><strong>@plroebuck</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3892" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3892/hovercard">#3892</a>: Rework reporter tests (<a href="https://urls.greenkeeper.io/plroebuck"><strong>@plroebuck</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3872" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3872/hovercard">#3872</a>: Rename <code>--exclude</code> to <code>--ignore</code> and create alias (<a href="https://urls.greenkeeper.io/boneskull"><strong>@boneskull</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3963" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3963/hovercard">#3963</a>: Hide stacktrace when cli args are missing (<a href="https://urls.greenkeeper.io/outsideris"><strong>@outsideris</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3956" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3956/hovercard">#3956</a>: Do not redeclare variable in docs array example (<a href="https://urls.greenkeeper.io/DanielRuf"><strong>@DanielRuf</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3957" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3957/hovercard">#3957</a>: Remove duplicate line-height property in <code>mocha.css</code> (<a href="https://urls.greenkeeper.io/DanielRuf"><strong>@DanielRuf</strong></a>)</li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/issues/3960" data-hovercard-type="pull_request" data-hovercard-url="/mochajs/mocha/pull/3960/hovercard">#3960</a>: Don't re-initialize grep option on watch re-run (<a href="https://urls.greenkeeper.io/geigerzaehler"><strong>@geigerzaehler</strong></a>)</li>
</ul>
</details>

<details>
<summary>Commits</summary>
<p>The new version differs by 39 commits.</p>
<ul>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/bd477769d6d76f467363ef2ff27e0b622aa5a132"><code>bd47776</code></a> <code>Release v6.2.0</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/cc595afaffc8117d14ebdd382018754d6c48c4b0"><code>cc595af</code></a> <code>update CHANGELOG.md for v6.2.0 [ci skip]</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/59d70ee1dfb29541a30d886ae152401dc7a35156"><code>59d70ee</code></a> <code>fix: remove duplicate line-height property (#3957)</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/f77cac46879426c924e947c749f57c72c59c3b5e"><code>f77cac4</code></a> <code>fix: do not redeclare variable (#3956)</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/6201e426bb6856f4dc3314fffb2f25fca719f12f"><code>6201e42</code></a> <code>Hide stacktrace when cli args are missing (#3963)</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/88f45d5fd146a03f728788c93594958f22942c44"><code>88f45d5</code></a> <code>Don't re-initialize grep option on watch re-run (#3960)</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/5d4dd98747637d0e7ed3007328ec9627dd7eda41"><code>5d4dd98</code></a> <code>Fix No Files error when file is passed via --files (#3942)</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/15b96afccaf508312445770e3af1c145d90b28c6"><code>15b96af</code></a> <code>Collect test files later (#3953)</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/ccee5f1b37bb405b81814daa35c63801cad20b4d"><code>ccee5f1</code></a> <code>Base reporter store ref to console.log (#3725)</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/47318a75fb0a218f843289de0511e8558f8b4df0"><code>47318a7</code></a> <code>update @mocha/contributors to v1.0.4 (#3944)</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/c903147f059fd5c8a0b2b61a47e9612fd4426760"><code>c903147</code></a> <code>More, improved integration tests for watching (#3929)</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/e341ea4ad9f02b2f55798872f49e15ee023b3134"><code>e341ea4</code></a> <code>Update CI config files to use Node-12.x (#3919)</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/3064d25ee50cde6aabeca5751e090c8893cdbd31"><code>3064d25</code></a> <code>update @mocha/docdash to v2.1.1 (#3945)</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/9ea45e7557a193c4b69816085397a6655fb5bc06"><code>9ea45e7</code></a> <code>do not fork if no node flags present (#3827)</code></li>
<li><a href="https://urls.greenkeeper.io/mochajs/mocha/commit/d02a0966d03b0945c77412bee5ea56b48ca26cf2"><code>d02a096</code></a> <code>modify Mocha constructor to accept options.global or options.globals (#3914)</code></li>
</ul>
<p>There are 39 commits in total.</p>
<p>See the <a href="https://urls.greenkeeper.io/mochajs/mocha/compare/31c019e0fffb4e77b7191ac5882edbe15f581c0b...bd477769d6d76f467363ef2ff27e0b622aa5a132">full diff</a></p>
</details>

<details>
<summary>FAQ and help</summary>

There is a collection of [frequently asked questions](https://greenkeeper.io/faq.html). If those don’t help, you can always [ask the humans behind Greenkeeper](https://github.com/greenkeeperio/greenkeeper/issues/new).
</details>

---

Your [Greenkeeper](https://greenkeeper.io) Bot :palm_tree:

### PR DESCRIPTION
**Pull request by andrewhughes101**
#13 removed a function that now means ping check promise doesnt resolve and the test cases timeout.
Updated expected test result to the correct error message
fixes #16

### CODE REVIEW

**Comment by andrewhughes101** [comment]
Reverted last commit as the error message is only different in node 12 and the original message is valid for node 8 and 10.

File: package.json [hunk 1]
```diff
@@ -22,13 +22,13 @@
...
-    "typescript": "^3.3.3333",
     "@types/chai": "^4.0.5",
     "@types/mocha": "^5.2.6",
     "@types/node": "^12.0.0",
     "chai": "^4.1.2",
+    "coveralls": "^3.0.2",
+    "mocha": "^6.2.0",
     "nyc": "^14.1.0",
-    "mocha": "^6.0.2"
+    "typescript": "^3.3.3333"
```
Comment (BethGriggs) [inline package.json]: Maybe try going to the latest TypeScript version.

Comment (andrewhughes101) [reply package.json]: Updated typescript and tests all pass

**Comment by andrewhughes101** [comment]
Updated packages @types/node , coveralls, @types/chai and all tests pass

refs #17 , #18 , #19

**Review by BethGriggs — APPROVED** [review]

**Pull request merged**